### ISSUE TITLE
How to retrieve Seller Profile information from Selling Partner API?

### ISSUE THREAD

**Issue by wolendranh** [amzn/selling-partner-api-docs#492]
According to documentation only thing that is related to Seller profile is `selling_partner_id`. For Advertising API it is possible to retrieve Profile, how can I do something similar for Selling Partner API?

**Comment by krithic15** [comment]
Hello there,

Currently we do not offer an API to pull Seller information. Apologies for the inconvenience. If you could provide details on your use-case, I can help find the best solution.

Best,
Krithi
Solutions Architect, SP API

**Comment by SebastianOzdoba** [comment]
Hi,
We want to show in our application information about configured seller account (name of account, not partner_id)

IMO seller-api is most worst api I have been working with (poor support (waiting for PII approve for months, cannot delete apps from developer profile), poor documentation, lack of sandbox)

**Comment by krithic15** [comment]
Hi @SebastianOzdoba,

The seller registration on your application's website is required for the oauth flow so that they are able to log in to your website and then have your application generate a state token to uniquely identify seller and tie the oauth code with. Hence, in your application registration process, you could have the seller provide their name.

Apologies for the inconvenience caused. We are continuously working on improving our processes and documentation and we appreciate your feedback. 
For sandbox support, please see https://github.com/amzn/selling-partner-api-docs/blob/main/guides/developer-guide/SellingPartnerApiDeveloperGuide.md#the-selling-partner-api-sandbox

**Comment by SebastianOzdoba** [comment]
@krithic15 this is joke not the sandbox, put JSON in swagger ;) 
Sandbox enviroment has UI, i can do the whole process, registrater customer, buy, simulating payments, processing orders etc.
Look this is a sandbox: https://allegro.pl.allegrosandbox.pl/

You don't understand, i want to show in my application configured seller login. For example my customer has Amazon login: ABC, when authorize in my App, i want to get seller profile and show it in my app.

**Comment by Mitalee** [comment]
> Hi @SebastianOzdoba,
> The seller registration on your application's website is required for the oauth flow so that they are able to log in to your website and then have your application generate a state token to uniquely identify seller and tie the oauth code with. Hence, in your application registration process, you could have the seller provide their name.

We need atleast an email ID, with which to process the user and log them in. LWA provides `https://api.amazon.com/user/profile`. Could you try and add this to the scopes list for SPAPI?

It's virtually impossible to integrate this for regular use without having an email id, which is crucial to add for an existing solution.

**Comment by AmichayMiara** [comment]
as @Mitalee said, LWA support getting user profile data. 
Please enable adding the scope "profile" to the authorization request.

**Comment by tspicer** [comment]
I concur with the prior two posts as the LWA `profile`  is something we leverage with the Amazon Advertising API. It was a surprise that it was also not present here.

**Comment by openbridge-james** [comment]
This is an important missing feature.  We work with many Oauth APIs, and they all work generally the same we request the user_id/profile_profile of the profile to ensure that when a customer grants us authorization we can decide if we need to create a new record, or update an existing record.  Since SPAPI doesn't provide the profile scope for LWA we have no means to make this decision.   Having a customer add an email address or something similar is a broken idea.  This should be defined by the LWA authorizing system so that customers can't make mistakes....

**Comment by krithic15** [comment]
Hello,

Thank you for your feedback. I have forwarded the feature enhancement request to our product team. We appreciate your patience!

Best,
Krithi
Solutions Architect, SP API

**Comment by hemangmehta** [comment]
@krithic15 any update on this? it's been a long time we are waiting for this. please let us know how long it will take to add profile scope?

**Comment by wolendranh** [comment]
@krithic15 Any update on this one? is there a way to access some broader information that `selling_partner_id`?

**Comment by moltar** [comment]
Not stale.

[GitHub stale bot considered harmful](https://drewdevault.com/2021/10/26/stalebot.html)

**Comment by Mitalee** [comment]
We would like to see if this is added.. it’s causing extra steps in the user register process which adds to customer support costs

**Comment by VijayaLakshmi210800** [comment]
@krithic15 any updates on this?

**Comment by mudassirzr** [comment]
@krithic15 this is much needed, as this can eliminate a lot in the user registration process within our app? Will this be implemented anytime soon?

**Comment by VijayaLakshmi210800** [comment]
@krithic15 yes please let us know are you planning for this feature or not?

**Comment by andreyAtBB** [comment]
Would be wonderful if SP API provided an ability to retrieve profile info like you do for Advertising API. Thanks!

**Comment by ericblade** [comment]
> Hello there,
> 
> Currently we do not offer an API to pull Seller information. Apologies for the inconvenience. If you could provide details on your use-case, I can help find the best solution.
> 
> Best, Krithi Solutions Architect, SP API

Several SPAPI calls require the SellerID as an input to them, but there's no way to obtain it.

**Comment by mukeshpiplai** [comment]
Yes, there is a tricky way to get SellerID using Amazon SP API.

You can use '**getMyFeesEstimateForASIN**' API endpoint under '**PRODUCT FEES API**' ie POST **/products/fees/v0/items/{Asin}/feesEstimate**.

link: [](https://developer-docs.amazon.com/sp-api/docs/product-fees-api-v0-reference#getmyfeesestimateforasin)

You can make a POST request for any ASIN like:

```
{
  "FeesEstimateRequest": {
    "MarketplaceId": "ATVPDKIKX0DER",
    "IsAmazonFulfilled": true,
    "PriceToEstimateFees": {
      "ListingPrice": {
        "CurrencyCode": "USD",
        "Amount": 10
      },
      "Shipping": {
        "CurrencyCode": "USD",
        "Amount": 0
      }
    },
    "Identifier": "B08WJ81ZS1"
  }
}
```
and you will get an API Response containing 'SellerId' for the used refresh token in that API call. Here is an example of an API response:

```
{
  "payload": {
    "FeesEstimateResult": {
      "Status": "ClientError",
      "FeesEstimateIdentifier": {
        "MarketplaceId": "ATVPDKIKX0DER",
        "IdType": "ASIN",
        "SellerId": "A2VYDBAL8BMVQK",
        "SellerInputIdentifier": "B08WJ81ZS1",
        "IsAmazonFulfilled": true,
        "IdValue": "B08WJ81ZS1",
        "PriceToEstimateFees": {
          "ListingPrice": {
            "CurrencyCode": "USD",
            "Amount": 10
          },
          "Shipping": {
            "CurrencyCode": "USD",
            "Amount": 0
          }
        }
      },
      "Error": {
        "Type": "Sender",
        "Code": "InvalidParameterValue",
        "Message": "There is an client-side error. Please verify your inputs.",
        "Detail": [
          
        ]
      }
    }
  }
}
```
Check the following JSON element in the above response: "payload.FeesEstimateResult.FeesEstimateIdentifier.SellerId"

It will work for Private and Public Developer (Both)

You can confirm that Seller Name and address using this link with the above sellerId: https://www.amazon.com/sp?seller=A2VYDBAL8BMVQK

See my answer here : https://stackoverflow.com/questions/75822171/is-there-any-endpoint-in-amazon-sp-api-to-get-seller-id

**Comment by ericblade** [comment]
I think the intended way is to have the user provide it when they make a request.  Saving the one that comes from the oauth response on the user's end, and then passing it to Amazon when needed seems to be The Way.

but there probably should be an API call that is intended to return "who am i making requests for"

**Comment by chazmcgarvey** [comment]
Be advised that there is a security hazard using `selling_partner_id` from the SP-API auth flow. This value is provided to the front end as a query parameter to the redirect URI, like `spapi_oauth_code`, but unlike `spapi_oauth_code` there's no documented way to verify your application back end receives the correct value. The value could be forged.

If you send `selling_partner_id` to your back end and it naively trusts the value, then your application is open to attack, the severity of which depends on how you use `selling_partner_id` in your back end.
